modules/renter/hostdb/hosttree/hosttree_test.go

```diff
@@ -160,6 +161,33 @@ func TestHostTree(t *testing.T) {
 	}
 }
 
+// Verify that inserting and fetching in parallel from the hosttree does not
+// cause a data race. We rely on the race detector for this test.
+func TestHostTreeParallel(t *testing.T) {
+	tree := New()
+
+	go func() {
+		time.Sleep(time.Millisecond * 4)
+		treeSize := 100
+		var keys []types.SiaPublicKey
+		for i := 0; i < treeSize; i++ {
+			entry := makeHostEntry(types.NewCurrency64(20))
+			keys = append(keys, entry.PublicKey)
+			err := tree.Insert(entry)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}()
+	go func() {
+		time.Sleep(time.Millisecond * 5)
+		_, err := tree.Fetch(2, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+}
```

Comment:
My preferred approach:

1. have a consistency check that can easily identify inconsistencies in the state. I believe we already have one of those for the host tree.

2. spin up like 100 gothreads in a for loop using a thread group. Have the gothreads running on an infinite loop and randomly either inserting, fetching, modifying, or using one of the other exported functions. You can use a map internal to the thread so it knows if it has inserted something before (therefore can fetch, modify, or delete that thing).

3. Let it run for a few seconds

4. tg.Stop() -> run the inconsistency check.

That will pretty much tell you if it's thread safe or not. This I feel does not get as much code coverage.